modules/tracktion_engine/model/automation/tracktion_AutomationCurve.cpp

```diff
@@ -772,24 +778,24 @@ CurvePoint AutomationCurve::getBezierHandle (int index) const noexcept
 
         auto bp = getBezierPoint (index);
 
-        auto x = getBezierXfromT (0.5, x1end, bp.time, x2end);
-        auto y = getBezierYFromX (x, x1end, y1end, bp.time, bp.value, x2end, y2end);
-        return { x, y };
+        auto x = getBezierXfromT (0.5, x1end, toTime (bp.time, getOwnerParameter()->getEdit().tempoSequence).inSeconds(), x2end);
+        auto y = getBezierYFromX (x, x1end, y1end, toTime (bp.time, getOwnerParameter()->getEdit().tempoSequence).inSeconds(), bp.value, x2end, y2end);
+        return { TimePosition::fromSeconds (x), y };
     }
 
     double x1end, x2end;
     float y1end, y2end;
     getBezierEnds (index, x1end, y1end, x2end, y2end);
-    return { x1end, y1end };
+    return { TimePosition::fromSeconds (x1end), y1end };
 }
 
 void AutomationCurve::getBezierEnds (int index, double& x1out, float& y1out, double& x2out, float& y2out) const noexcept
 {
-    auto x1 = getPointTime (index);
+    auto x1 = getPointTime (index).inSeconds();
```

Comment:
would it be better to make the function return position objects instead of doubles?

Reply:
Yes... I do remember thinking that at the time but decided against it. But I can't remember why now. Probably because when I started I was trying to get it to build with minimal changes and then later abandoned that idea as it was more tricky to go half in. I'll update that.